python/mxnet/kvstore.py

```diff
@@ -35,6 +35,32 @@ def _ctype_key_value(keys, vals):
             c_vals += c_val_i
         return (c_array(ctypes.c_int, c_keys), c_array(NDArrayHandle, c_vals))
 
+def _ctype_str_key_value(keys, vals):
+    names = []
+    if isinstance(keys, str):
+        if isinstance(vals, NDArray):
+            names.append(c_str(keys))
+            return (c_array(ctypes.c_char_p, names),
+                    c_array(NDArrayHandle, [vals.handle]))
+        else:
+            for value in vals:
+                assert(isinstance(value, NDArray))
+            return (c_array(ctypes.c_char_p, [c_str(keys)] * len(vals)),
```

Comment:
why? I think when vals is a list of NDArray, it means they are duplicates of the same weight on multiple devices